lib/backend/backend.go

```diff
@@ -92,6 +92,25 @@ type Backend interface {
 	CloseWatchers()
 }
 
+// Migration is an interface responsible for applying data migrations to the backend.
+//
+// Only a single migration should be run at once. Migrations are applied in sequential
+// order based on their Version. No two migrations should have the same Version, and there
+// should be no holes in the Version sequence.
+type Migration interface {
```

Comment:
Looking at it, I don't see any calls that would make it need to be an exported interface - unless there's plans for this in the future. So perhaps this could be in `migration` and be unexported, so the stutter wouldn't be a concern.